s3-handler/main.go

```diff
@@ -108,6 +111,22 @@ func Handler(request events.S3Event) (Response, error) {
 				parseXRayTraceID(tmp.(string), parsedLine)
 			}
 
+			if req, ok := parsedLine["request"].(string); redactPattern != nil && ok {
+				parsedLine["request"] = redactRequest(req, redactPattern)
+			}
+
+			if req, ok := parsedLine["request"].(string); ok && parserType == "alb" {
+				httpMeta, err := parseRequestHttpMeta(req)
+				if err != nil {
+					logrus.WithError(err).WithField("request", req).
+						Error("failed to parse request metadata")
+				} else {
+					for k, v := range httpMeta {
+						parsedLine[k] = v
+					}
+				}
+			}
```

Comment:
We only need to parse the request string once and reuse for both checks.

```suggestion
			if req, ok := parsedLine["request"].(string); ok {
				if redactPattern != nil {
					parsedLine["request"] = redactRequest(req, redactPattern)
				}

				if parserType == "alb" {
					httpMeta, err := parseRequestHttpMeta(req)
					if err != nil {
						logrus.WithError(err).WithField("request", req).
							Error("failed to parse request metadata")
					} else {
						for k, v := range httpMeta {
							parsedLine[k] = v
						}
					}
				}
			}
```